src/core/reactor.cc

```diff
@@ -1195,7 +1195,11 @@ cpu_stall_detector_linux_perf_event::~cpu_stall_detector_linux_perf_event() {
 void
 cpu_stall_detector_linux_perf_event::arm_timer() {
     uint64_t ns = (_threshold * _report_at + _slack) / 1ns;
+    #if __cplusplus >= 202002L
```

Comment:
This preprocessor switch should be outside of .cc file and define some easy to use macro (like [this](https://github.com/scylladb/seastar/blob/master/include/seastar/util/concepts.hh#L23)).
Another option is to replace the `[[likely]]` with `__builtin_expect`

Reply:
make sense, let me update this part